core-api/src/main/java/com/optimizely/ab/notification/DecisionNotification.java

```diff
@@ -0,0 +1,113 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package com.optimizely.ab.notification;
+
+
+import com.optimizely.ab.bucketing.FeatureDecision;
+import com.optimizely.ab.config.FeatureVariable;
+import com.optimizely.ab.config.Variation;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DecisionNotification {
+    protected String type;
+    protected String userId;
+    protected Map<String, ?> attributes;
+    protected Map<String, ?> decisionInfo;
+
+    protected DecisionNotification() {
+    }
+
+    protected DecisionNotification(@Nonnull String type,
+                                   @Nonnull String userId,
+                                   @Nullable Map<String, ?> attributes,
+                                   @Nonnull Map<String, ?> decisionInfo) {
+        this.type = type;
+        this.userId = userId;
+        if (attributes == null) {
+            attributes = new HashMap<>();
+        }
+        this.attributes = attributes;
+        this.decisionInfo = decisionInfo;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public Map<String, ?> getAttributes() {
+        return attributes;
+    }
+
+    public Map<String, ?> getDecisionInfo() {
+        return decisionInfo;
+    }
+
+    public static ExperimentDecisionNotificationBuilder newExperimentDecisionNotificationBuilder() {
+        return new ExperimentDecisionNotificationBuilder();
+    }
+
+    public static class ExperimentDecisionNotificationBuilder {
+        public final static String EXPERIMENT_KEY = "experiment_key";
+        public final static String VARIATION_KEY = "variation_key";
+
+        private String experimentKey;
+        private Variation variation;
+        private String userId;
+        private Map<String, ?> attributes;
+        private Map<String, Object> decisionInfo;
+
+        public ExperimentDecisionNotificationBuilder withUserId(String userId) {
+            this.userId = userId;
+            return this;
+        }
+
+        public ExperimentDecisionNotificationBuilder withAttributes(Map<String, ?> attributes) {
+            this.attributes = attributes;
+            return this;
+        }
+
+        public ExperimentDecisionNotificationBuilder withExperimentKey(String experimentKey) {
+            this.experimentKey = experimentKey;
+            return this;
+        }
+
+        public ExperimentDecisionNotificationBuilder withVariation(Variation variation) {
```

Comment:
I feel this should be `withVariationKey`. Just seems inconsistent.

Reply:
Actually I am passing variation here so that on build we can verify null check before getting key from its object